### PR TITLE
Migration of all blocks to their own files

### PR DESCRIPTION
**Pull request by ghost**
As per the ACL block:
#253

### CODE REVIEW

File: fastly/block_fastly_service_v1_logentries.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Where in the generated VCL the logging call should be placed.",
+				ValidateFunc: validateLoggingPlacement(),
+			},
+		},
+	},
+}
+
+func processLogEntries(d *schema.ResourceData, conn *gofastly.Client, latestVersion int) error {
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logentries.go]: nitpick - 
```suggestion
func processLogentries(d *schema.ResourceData, conn *gofastly.Client, latestVersion int) error {
```

File: fastly/block_fastly_service_v1_logentries.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+		_, err := conn.CreateLogentries(&opts)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func readLogEntries(conn *gofastly.Client, d *schema.ResourceData, s *gofastly.ServiceDetail) error {
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logentries.go]: nitpick - 
```suggestion
func readLogentries(conn *gofastly.Client, d *schema.ResourceData, s *gofastly.ServiceDetail) error {
```

File: fastly/block_fastly_service_v1_director.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+	directorList, err := conn.ListDirectors(&gofastly.ListDirectorsInput{
+		Service: d.Id(),
+		Version: s.ActiveVersion.Number,
+	})
+
+	if err != nil {
+		return fmt.Errorf("[ERR] Error looking up Directors for (%s), version (%v): %s", d.Id(), s.ActiveVersion.Number, err)
+	}
+
+	log.Printf("[DEBUG] Refreshing Backends for (%s)", d.Id())
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_director.go:167]: I like that you removed passing in the list of backends and keep everything contained in this function.

commit - https://github.com/opencredo/terraform-provider-fastly/pull/43/commits/c4a6c3c41be56a8348443ee1d950073b8e1fa9eb

**Review by mccurdyc — COMMENTED** [review]
One super minor nitpick for `LogEntries` -> `Logentries`. Looks great!

I see that CI is complaining about a `make fmt` though.

:smile: - https://github.com/opencredo/terraform-provider-fastly/pull/43/commits/ce3c5f6877dd228f8799beb600829e6fd9809cfd

I saw that you removed, which is fine, I have a PR to add it back (https://github.com/terraform-providers/terraform-provider-fastly/pull/259)

It's obvious that this new pattern makes it easier to identify bugs.

**Review by mccurdyc — APPROVED** [review]

File: fastly/block_fastly_service_v1_logentries.go (hunk 2 above)
Comment (ghost) [reply fastly/block_fastly_service_v1_logentries.go]: This has been updated

File: fastly/block_fastly_service_v1_director.go (hunk 3 above)
Comment (ghost) [reply fastly/block_fastly_service_v1_director.go:167]: Thanks

File: fastly/block_fastly_service_v1_logentries.go (hunk 2 above)
Comment (ghost) [reply fastly/block_fastly_service_v1_logentries.go]: We've uncommented the ReadElasticSearch and ReadFTP functions as per your PR

**Review by phamann — APPROVED** [review]
:+1: LGTM - After reading the `resource_service_v1` logic and a run of the acceptance tests.

**Pull request merged**